File: 08. Trees/03. Postorder Traversal/main.js
class TreeNode {
    constructor(val) {
        this.val = val;
        this.left = this.right = null;
    }
}

// Postorder Traversal (Recursive)
function postorderTraversal(root, result) {
    if (!root) return;
    postorderTraversal(root.left, result);
    postorderTraversal(root.right, result);
    result.push(root.val);
}

// Main function
function main() {
    // Sample tree
    let root = new TreeNode(1);
    root.left = new TreeNode(2);
    root.right = new TreeNode(3);
    root.left.left = new TreeNode(4);
    root.left.right = new TreeNode(5);
    root.right.left = new TreeNode(6);
    root.right.right = new TreeNode(7);

    let result = [];
    postorderTraversal(root, result);
    console.log("Postorder Traversal:", result);
}

// Run the main function
main();

File: 08. Trees/03. Postorder Traversal/main.py
class TreeNode:
    def __init__(self, val=0):
        self.val = val
        self.left = None
        self.right = None

# Postorder Traversal (Recursive)
def postorder_traversal(root, result):
    if not root:
        return
    postorder_traversal(root.left, result)
    postorder_traversal(root.right, result)
    result.append(root.val)

# Main function
def main():
    # Sample tree
    root = TreeNode(1)
    root.left = TreeNode(2)
    root.right = TreeNode(3)
    root.left.left = TreeNode(4)
    root.left.right = TreeNode(5)
    root.right.left = TreeNode(6)
    root.right.right = TreeNode(7)

    result = []
    postorder_traversal(root, result)
    print("Postorder Traversal:", result)

# Run the main function
main()
